server/app/views/applicant/ProgramCardViewRenderer.java

```diff
@@ -0,0 +1,340 @@
+package views.applicant;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static j2html.TagCreator.a;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.h3;
+import static j2html.TagCreator.h4;
+import static j2html.TagCreator.img;
+import static j2html.TagCreator.li;
+import static j2html.TagCreator.p;
+import static j2html.TagCreator.span;
+import static j2html.TagCreator.text;
+import static services.applicant.ApplicantPersonalInfo.ApplicantType.GUEST;
+import static views.applicant.AuthenticateUpsellCreator.createLoginPromptModal;
+import static views.components.Modal.RepeatOpenBehavior.Group.PROGRAMS_INDEX_LOGIN_PROMPT;
+
+import auth.CiviFormProfile;
+import auth.ProfileUtils;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import controllers.applicant.ApplicantRoutes;
+import j2html.TagCreator;
+import j2html.tags.ContainerTag;
+import j2html.tags.DomContent;
+import j2html.tags.specialized.ATag;
+import j2html.tags.specialized.DivTag;
+import j2html.tags.specialized.ImgTag;
+import j2html.tags.specialized.LiTag;
+import j2html.tags.specialized.PTag;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import javax.inject.Inject;
+import play.i18n.Messages;
+import play.mvc.Http;
+import services.MessageKey;
+import services.applicant.ApplicantPersonalInfo;
+import services.applicant.ApplicantService;
+import services.cloud.PublicFileNameFormatter;
+import services.cloud.PublicStorageClient;
+import services.program.ProgramDefinition;
+import services.program.StatusDefinitions;
+import services.settings.SettingsManifest;
+import views.HtmlBundle;
+import views.TranslationUtils;
+import views.components.ButtonStyles;
+import views.components.Icons;
+import views.components.LinkElement;
+import views.components.Modal;
+import views.components.TextFormatter;
+import views.style.ApplicantStyles;
+import views.style.BaseStyles;
+import views.style.ReferenceClasses;
+
+/** A renderer to create an individual program information card. */
+public final class ProgramCardViewRenderer {
```

Comment:
praise:  Thanks for making this reusable!  I also have a feeling that splitting this out could help us when we migrate to a templating tool and split the markup into potentially "hot-swappable" chunks.